### ISSUE TITLE
[Question] [`FuncEnv`] `params` are not used by `__init__` & `render_init` & `render_close`

### ISSUE THREAD

**Issue by Kallinteris-Andreas** [Farama-Foundation/Gymnasium#861]
### Question

- `__init__` uses `options` instead of `params`
- `render_init` uses `kwargs` instead of `params`
- `render_close` uses nothing (granted, I do not see a need to for it to include `params`)

**Comment by RedTachyon** [comment]
This is somewhat similar to #851, so I'll address them together.

Basically, there's a tradeoff between being consistent and being efficient. In case of the RNG argument, if a certain function isn't expected to use RNG, we can omit it for brevity, or include it for consistency (and for the edge cases that might use it)

I think in both cases, I tried to be a bit too clever going for efficiency, and it might instead be better to keep it consistent instead - every single method would be expected to have both `rng` and `params`? This seems a bit weird for things like `render_close`, so I'm not sure. But we definitely should have some clear principles, e.g. maybe omit `rng` just for rendering and info-related functions? (which are all supposed to convey some sort of information about the environment "from the outside")

As for the specific questions in this issue - `__init__` options serve a different purpose than `params` everywhere else, so it's good imo to keep them named differently. `render_init` initially had `**kwargs` because it's a thing that's unlikely to be jitted anyways, plus it's hard to predict what users might want to put there, but with `params` introduced, it's probably better to just use that. `render_close` probably doesn't need anything, but we should keep it consistent. And it might need something, e.g. we know if we need to close a pygame window based on whether it was created (based on the params)

**Comment by Kallinteris-Andreas** [comment]
When we are making an Environment API, it should support all environments (other than perhaps very weird edge cases) so:

## rng
- `observation` & `reward` `terminal` should all have an `rng` argument, which, even if rare, some environments may need them
- `transition_info` & `state_info` should not have one, because they are deterministic by nature
- `render_image` & `render_init` `render_close` should not have one, as they should be a deterministic rendering (it would be worrying if an environment needs to present nondeterministic information to the user)
- `transform` & `get_default_params` unless you are psycho, they should be deterministic

## params
- `render_init` `**kwargs` should be replaced with `params`
- `render_close` add `params`

**Comment by RogerJL** [comment]
I have no need of `rng` in `observation`, completely deterministic, and calls to `split` function to generate new are among the slowest in my loop.

state - global state (`transform` could split new `rngs` saved here if needed)

With that `rng` would only be needed to initial

**Comment by Kallinteris-Andreas** [comment]
`observation` is not completely deterministic, in fact it is never deterministic in real world problems, real world sensors always have some noise with them

**Comment by RogerJL** [comment]
In **my** project `observation` is completely deterministic - playing Ludo.
Unless you want to restrict usage of Gymnasium to simulating physical systems.

I only need randomness when `transforming` state(board), action -> state(new board, next player, die rolled)
Using `split` for one rng to only roll one die is a bit wasteful, state could contain any number of dices and a rng to generate next set. And worse is doing a spit creating an rngs that is not even used...

**Comment by Kallinteris-Andreas** [comment]
1. APIs should be designed to support all possible (or at least as many as possible environments)
2. Can you make a performance measurement of the impact of those extra `rng` arguments, I can not possibly imagine it being too much

**Comment by RogerJL** [comment]
1. If something is added to an API before 1.0 you will have to live with it almost forever...
2. Sadly the Perfetto trace coverage stopped working for me.

Done some testing, looks like what matters most is if code is `split` is inside or outside of `jax.jit` compiled code.
If the main loop is in a jitted function it should not matter when and where splits are made.
